doc/protocol.md

```diff
@@ -355,3 +355,62 @@ errors, which means that the same fragment could very well be received multiple
 Relays may forward fragments even if they arrive out of order.
 
 
+# Common Encoding For Media over QUIC Streams
+
```

Comment:
I am not sure that we should have a section on "common encodings for streams", by opposition to just adding section for RUSH and WARP, and updating the architecture part to indicate that we support 4 encodings: single stream, WARP, RUSH, Datagrams. Define how the 'subscribe" and "post" messages indicate these types. Then have a section for each of those types.

Reply:
I feel the subscribe/post message shouldn't carry a particular grouping in them. The common format ensure you get some data that can be related to a mediaId, groupId and objectId. If it comes in one stream (for all the media - QUICR), one stream per gop (warp), one stream per frame (rush) , it can still group things as we do today

Reply:
The receiver can certainly be ready to receive data in any format. However, when we start an experiment, it is nice to have an easy way to specify whether we expect data in one or another format. Plus, we may well have some future clients that can have only implemented a subset of the transmission types...

Also, currently the subscribe messages only carry the mediaId (or datagram stream ID) when the client selects the "datagram" mode -- but that one is an easy fix, just send the media ID all the time.

Reply:
I see what you mean here. The scope here is quicrq prototype implementation, but are you ok with this general direction if i do a PR for the ietf draft ..